lib/collector/api.js

```diff
@@ -68,20 +68,24 @@ function CollectorAPI(agent) {
   /* RemoteMethods can be reused and have little per-object state, so why not
    * save some GC time?
    */
-  this._methods = {
-    preconnect: new RemoteMethod('preconnect', agent.config, initialEndpoint),
-    connect: new RemoteMethod('connect', agent.config, initialEndpoint),
-    settings: new RemoteMethod('agent_settings', agent.config, initialEndpoint),
-    errors: new RemoteMethod('error_data', agent.config, initialEndpoint),
-    metrics: new RemoteMethod('metric_data', agent.config, initialEndpoint),
-    traces: new RemoteMethod('transaction_sample_data', agent.config, initialEndpoint),
-    shutdown: new RemoteMethod('shutdown', agent.config, initialEndpoint),
-    events: new RemoteMethod('analytic_event_data', agent.config, initialEndpoint),
-    customEvents: new RemoteMethod('custom_event_data', agent.config, initialEndpoint),
-    queryData: new RemoteMethod('sql_trace_data', agent.config, initialEndpoint),
-    errorEvents: new RemoteMethod('error_event_data', agent.config, initialEndpoint),
-    spanEvents: new RemoteMethod('span_event_data', agent.config, initialEndpoint),
-    logEvents: new RemoteMethod('log_event_data', agent.config, initialEndpoint)
+  this._methods = {}
+  for (const [accessor, name] of [
+    ['preconnect', 'preconnect'],
+    ['connect', 'connect'],
+    ['settings', 'agent_settings'],
+    ['errors', 'error_data'],
+    ['metrics', 'metric_data'],
+    ['traces', 'transaction_sample_data'],
+    ['shutdown', 'shutdown'],
+    ['events', 'analytic_event_data'],
+    ['customEvents', 'custom_event_data'],
+    ['queryData', 'sql_trace_data'],
+    ['errorEvents', 'error_event_data'],
+    ['spanEvents', 'span_event_data'],
+    ['logEvents', 'log_event_data']
+  ]) {
+    const method = new RemoteMethod(name, agent.config, initialEndpoint, agent.metrics)
```

Comment:
this won't work. I noticed it while testing with a real app and not seeing these new output bytes metrics.  If you test for yourself and put a break point here `agent.metrics` is undefined.  This is because the collector instance is created before the metrics aggregator: https://github.com/newrelic/node-newrelic/blob/main/lib/agent.js#L77

Reply:
Welp, that's frustrating. I guess we might need to pass the whole agent to the remote methods after all, to preserve that reference rather than passing undefined by value?

Reply:
I think passing in the whole agent is the easiest thing to do. I started playing around trying some other more sophisticated things but I couldn't figure out how to do it simply.